Replaced Servlet DSL with Revenj.

zapov opened the pull request:
Added DB tests.
Both Java and C# version.

zapov commented:
Something wrong with Mono?
https://travis-ci.org/TechEmpower/FrameworkBenchmarks/jobs/98245070#L1981

blee-techempower commented:
Yes. Something was wrong with mono. I figure out that script was wrong, so I fixed it. Now it works, but I have not merge it to master branch.

zapov commented:
Previous Mono setup is broken.
Replaced it with latest stable Mono version + apt-get install.

msmith-techempower commented:
`toolset/setup/linux/languages/mono.sh` is unacceptable in its current state. If the mono installation is now working with the updated urls, please uncomment out the rest.

zapov commented:
Its not working in its current (before this commit) state. Whats unacceptable about it?

zapov commented:
Nor its working with just url update since that snapshot version does not exist on the server.

msmith-techempower commented:
That seems to be the problem - we are working on updating the mono installation to a more long-term url, but outside of that it should not change.

zapov commented:
I can add specific version to apt-get such as
sudo apt-get install mono-complete=4.2.1.102-0xamarin1
Would that be good enough?

msmith-techempower commented:
We have just merged in the fix for mono, but it means you will have to rebase your pull request.

zapov commented:
So I rebased it to upstream, but starting Mono failed with:

Could not load type 'Mono.Net.Security.Private.MonoTlsProviderImpl' from assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.
Setup Revenj.NET: Could not resolve type with token 0100004a
Setup Revenj.NET: Could not resolve type with token 0100003a
Setup Revenj.NET: Could not resolve type with token 0100003b ---> System.Reflection.ReflectionTypeLoadException: The classes in the module cannot be loaded.

I guess thats what you get when using snapshot versions of Mono.

msmith-techempower commented on toolset/setup/linux/languages/mono.sh:
```diff
@@ -1,52 +1,54 @@
+#mv $IROOT/mono.installing $IROOT/mono.installed
 
-source $IROOT/mono.installed
+#source $IROOT/mono.installed
```
This file needs to be reverted to our latest on master.

zapov replied:
I'm not sure you understood my last comment.
I rebased to your master (reverted this file) but starting Revenj with that snapshot version of Mono fails.
I included the exception message from it, so you can see using snapshot version of the Mono is not really a good solution.

msmith-techempower replied:
Understood, we will try and find a more permanent solution and get back to you.

zapov commented:
Any updates on Mono?

Why don't you change it to my proposed solution: apt-get install -y mono-complete=4.2.1.102-0xamarin1 ?

Do you want me to remove mono.sh from commit and push it as is (so it fails in Travis due to broken Mono snapshot version)?

zapov commented:
I've rebased this PR again to your latest master.
Removed the proper fix for mono.

Travis fails to download your specified snapshot version (so both fail in CI), but both frameworks pass tests when return 0 is inserted to mono.sh and any official version of the mono is installed.

Hopefully you'll stop experimenting with Mono and install it the same way as Java (or even better with specific version as I already suggested).

Regards,
Rikard

msmith-techempower commented:
@zapov Mono is in a state of disarray - let's see your suggested fix to `mono.sh` and I will reevaluate.

zapov commented:
You want me to add mono.sh with
sudo apt-get install mono-complete=4.2.1.102-0xamarin1
to this PR?

msmith-techempower commented:
Yes, let's just get Mono working and I'll deal with the repercussions later.

zapov commented:
There you go.

Btw. There is xsp dependency for some C# frameworks. You should remove that, since xsp should be included with mono-complete.

msmith-techempower commented:
Thanks.

Revenj passes all tests in Travis; LGTM!
